app/App.tsx

```diff
@@ -10,6 +11,7 @@ import lazyLoad from './utils/LazyLoad';
 import './assets/css/main';
 
 declare const System;
+declare const window: any;
```

Comment:
Extend the window interface here rather then stubbing out the entire object. Otherwise this could get dangerous down the road because you now have 0 type checking happening on `window`.

So, in `globals.d.ts`, do this...

```ts
interface Window {
    addThis: TheInterfaceOfTheAddThisGlobalHere;
}
```

This causes window to keep all its existing properties and also inherit whatever you set.

Reply:
Nice tip. Does the d.ts file automatically get recognized by the compiler as a definition file and applied to all sub-modules?

Reply:
As long as it exists somewhere in the project path (defined in tsconfig.json). In the case of `globals.d.ts`, it's at the root so it gets picked up. But it doesn't have to be at the root. It can be anywhere.